tests/api/printf.cpp

```diff
@@ -34,6 +34,9 @@ static std::string stdoutFileName;
 #define BUFFER_SIZE 1024
 static char stdoutBuffer[BUFFER_SIZE];
 
+std::vector<char> GLOBAL_BUFFER;
+size_t BUFFER_FILL_LEVEL = 0;
```

Comment:
I think we could get rid of this global variable.

Let's do it like that:
- Resize the global buffer to `0` at the beginning of each test.
- in printf_callback resize it to be able to copy the content